waku/v2/protocol/waku_store/protocol.nim

```diff
@@ -213,19 +225,26 @@ proc handleMessage*(w: WakuStore, pubsubTopic: string, msg: WakuMessage) =
     # The message is ephemeral, should not be stored
     return
   
+  if not isValidMessage(msg):
+    waku_store_errors.inc(labelValues = [invalidMessage])
```

Comment:
Not necessarily for this PR: I know this messes with the very clean boolean logic, but in practice messages usually arrive "late" or "early" for quite different reasons and it may be worth differentiating between these two invalid cases for debugging purposes. If the majority of messages arrive "late", this may be a case of network congestion. If the majority of messages arrive "early", it may be that our node is lagging, struggling to keep up or that you require an NTP sync,

Reply:
I did not think about that possibility, thanks for that. It is a very good point to differentiate why a message is invalid. We never get enough metrics about how things are performing.

I will add it to my TODO list and tackle it in one of the upcoming PRs as it is low-hanging fruit.